### DISCUSSION TITLE
Assistance Required for EmotiBit Device Issues

### DISCUSSION THREAD

**Discussion by LauraMMH** [EmotiBit/EmotiBit_Docs#129] [Q&A]
Hello!
I am writing to seek assistance regarding several issues we have been experiencing with the EmotiBit devices in our research setup.
We are currently utilizing the emotibit model that you can identify in the attached photograph for our research purposes. However, we have encountered problems with broken streams and missing data, particularly when attempting to send stimulus triggers to the recorder. We are using LabStreamingLayer as the interpreter between PsychoPy (version 2023v2.3) and the EmotiBit devices, but we are receiving constant error messages that I will attach as images for your reference.
To keep track of the streams, we have developed our PsychoPy code to pair each EmotiBit with a specific computer. We ensure that the stream is sent using the same router and network for consistency. However, despite these precautions, we are still facing challenges with data logging, and recording.
Could you please provide guidance on strategies to resolve these issues? Any insights or recommendations on how to improve the reliability of the data streams would be greatly appreciated.
Thank you for your attention.
![Imagen de WhatsApp 2024-10-21 a las 10 07 07_3fd1c565](https://github.com/user-attachments/assets/34a6c92e-cb3a-4af5-b206-281892247881)
![Imagen de WhatsApp 2024-10-21 a las 10 07 07_ecb4e0e4](https://github.com/user-attachments/assets/5570048c-1453-4235-b3fe-ed4cbe5e4524)
![Imagen de WhatsApp 2024-10-21 a las 10 07 08_7b8a29ae](https://github.com/user-attachments/assets/4e6db2e2-ca4d-4545-9cba-34826a977d19)
![Imagen de WhatsApp 2024-10-21 a las 10 07 08_c0813043](https://github.com/user-attachments/assets/fea00713-66b3-4ba4-89e5-90a26ea61922)
![Imagen de WhatsApp 2024-10-30 a las 09 39 29_370e5f3f](https://github.com/user-attachments/assets/cfec149c-d63f-457a-bd92-d68658e9c007)
![Imagen de WhatsApp 2024-10-30 a las 09 39 30_02e7854c](https://github.com/user-attachments/assets/081b5d74-bba4-422e-b397-9765c8c33514)
![Imagen de WhatsApp 2024-10-30 a las 09 39 30_3ef0e459](https://github.com/user-attachments/assets/99aa287f-6732-4542-8d1b-47578bf54103)
![Imagen de WhatsApp 2024-10-30 a las 09 39 30_5caa6001](https://github.com/user-attachments/assets/8ac748e1-3ad9-4ea7-83f6-256c02202724)
![Imagen de WhatsApp 2024-10-30 a las 09 39 30_7513f655](https://github.com/user-attachments/assets/890f87b6-2b92-4925-bf19-049c3d853f54)

Best regards.

**Comment by nitin710** [comment]
@LauraMMH Can you add more context to " particularly when attempting to send stimulus triggers to the recorder."?

Trying to understand if there is a particular action/activity that triggers a LSL break.

**Reply by LauraMMH** [reply]
Hi,
Thank you for your response. I hope this message provides a clearer description of the issues we are experiencing.

We are using EmotiBit devices to record peripheral activity during cognitive tasks presented on a computer. The interface is Psychopy. We have configured the device to send triggers from Psychopy to EmotiBit to mark the start and end of each task, allowing us to associate changes in physiological activation with specific task times.

Our setup involves multiple tasks performed in sequence, so we need to synchronize the task triggers (start and end) with the stimulus presentation. To achieve this, we use a start and end trigger for each task, sending these triggers via the LabStreamingLayer option in the Oscilloscope application. This allows the LabStreamingLayer plugin in Psychopy to identify and send the triggers for registration in EmotiBit.

However, we are experiencing the following issues:

Signal loss: Occasionally, the device stops recording, and we realize that no data is being captured.
Missing "LM" file in Data Parser output: The resulting file in the Data Parser does not contain the "LM" file, which should include the triggers.
Persistent noise in some EmotiBit devices: Some EmotiBit devices produce continuous noise (spikes in activity), even when they are not worn or when no movement is detected.

**Reply by LauraMMH** [reply]
Perhaps I should also mention that we are conducting several recordings simultaneously. For example, four users are working at the same desk, each using a different laptop to complete the cognitive tasks.

**Comment by nitin710** [comment]
I'll try to approach this from 2 sides, 1 that targets LSL markers and the other relating to the device issues.

## For LSL
- EmotiBit Oscilloscope receives the LSL markers from the LSL stream information provided in the `emotibitCommSettings.json` file. See our [documentation](https://github.com/EmotiBit/EmotiBit_Docs/blob/master/Working_with_emotibit_data.md#timesync-with-lsl-using-marker-stream) for more details. Can you confirm this file has been updated correctly to receive the appropriate triggers? You can also verify this on the bottom status bar when the Oscilloscope is running.
  - ![image](https://github.com/user-attachments/assets/169e0311-c218-4afa-b659-afd53aceeee8)
- Additonally, can you please share the Emotibit software version you are using?
- Note that to accomplish "_we use a start and end trigger for each task, sending these triggers via the LabStreamingLayer option in the Oscilloscope application._", **you don't need to toggle the LSL option(under the Output list)**. Simply specifying the trigger stream name as explained in the point above should accomplish adding LSL markers to the EmotiBit data.

## Relating to other device issues
- **Signal loss**:
  - _Occasionally, the device stops recording, and we realize that no data is being captured._
    - Can you confirm if you are referring to the device streaming data to the Oscillosocpe or the device recording data to the SD-Card (as indicated by the blinking RED led)?
    - Furthermore, can you add more details on this behavior? Does the device "reset" or simply continues to stream data to the Oscilloscope, but the RED recording LED stops blinking?
- **Missing "LM" file in Data Parser output** 
  - I will wait for your response in the above section before addressing this further.
- **Persistent noise in some EmotiBit devices**
  - Is it possible for you to share a screenshot of the Oscilloscope here, indicating the noise you are referring? I'm very curious to understand how "spikes in activity" is represented in the data. That might give us clues to understand where a potential problem may be coming from.
